lib/wnews.dart

```diff
@@ -1,48 +1,53 @@
-import 'package:xpath_selector_html_parser/xpath_selector_html_parser.dart';
+import 'package:html/parser.dart' as html;
 
-/// Takes the HTML contents of something like
-/// https://en.wikipedia.org/wiki/Main_Page
-/// and extracts the items in the _In the news_ section.
+/// Parses the JSON response from
+/// https://api.wikimedia.org/feed/v1/wikipedia/en/featured/2023/10/30
+/// where date is current day and language is parameter
+/// and extracts news items.
 ///
-/// For the sake of simplicity, this currently only works for the English
-/// Wikipedia, and assumes quite a lot about the structure
-/// of the page.
-List<NewsItem> extractNews(String html) {
-  final parser = HtmlXPath.html(html);
+/// For each item uses story and links parameters
+List<NewsItem> extractNews(String json) {
+  final parsedJson = jsonDecode(json) as Map<String, dynamic>;
+  final news = parsedJson['news'] as List<dynamic>?;
 
-  final newsHeadline =
-      parser.query("//h2[contains(text(),'In the news')]").node;
-  if (newsHeadline == null) {
-    throw ArgumentError('Cannot find the headline "In the news"');
+  if (news == null || news.isEmpty) {
+    throw StateError('The list is empty');
   }
 
-  final boxDiv = newsHeadline.queryXPath('/ancestor::div').node;
-  if (boxDiv == null) {
-    throw StateError('Cannot find the ancestor <div> of "In the news"');
-  }
+  final newsList = news.map((n) {
+    final currentNew = n['story'];
 
-  final firstList = boxDiv.queryXPath("/descendant::ul[1]").node;
-  if (firstList == null) {
-    throw StateError('Cannot find the first list of news');
-  }
+    // Parse the HTML content within the "story" field
+    final document = html.parse(currentNew);
 
-  final listItems = firstList.queryXPath('/li').nodes;
-  if (listItems.isEmpty) {
-    throw StateError('The list is empty');
-  }
+    // Extract the text content from the parsed HTML
+    final newsContent = document.body!.text;
 
-  NewsItem extract(XPathNode node) {
-    final text = node.text!.replaceAll(' (pictured)', '');
-    final linkElement = node.queryXPath('//b//a').node;
-    Uri? link;
-    if (linkElement != null) {
-      link = Uri.tryParse(linkElement.attributes['href'] ?? '');
+    // Extract url from json
+    final links = n['links'] as List<dynamic>?;
+    Uri? uri;
+    if (links != null && links.isNotEmpty) {
+      final url = links.first['content_urls']['desktop']['page'] as String?;
+      if (url != null) uri = Uri.tryParse(url);
     }
-    return NewsItem(text, link);
-  }
 
-  return listItems.map((li) => extract(li)).toList(growable: false);
+    return NewsItem(newsContent, uri);
+  }).toList(growable: false);
+
+  return newsList;
+}
+
+NewsItem extract(XPathNode node) {
+  final text = node.text!.replaceAll(' (pictured)', '');
+  final linkElement = node.queryXPath('//b//a').node;
+  Uri? link;
+  if (linkElement != null) {
+    link = Uri.tryParse(linkElement.attributes['href'] ?? '');
+  }
+  return NewsItem(text, link);
 }
```

Comment:
I don't think this is needed anymore? In that case, I think you can also drop the dependency on xpath.